.github/workflows/test.yml

```diff
@@ -7,9 +7,15 @@ on:
 env:
   GO111MODULE: on
 
+permissions:
+  contents: read
+
 jobs:
 
   golangci-lint:
+    permissions:
+      contents: read  # for actions/checkout to fetch code
```

Comment:
This seems a little redundant to set the per-job permissions the same as the overall setting above. Are these changes generated from [the security tool](https://github.com/step-security/secure-workflows#hosted-instance-appstepsecurityiosecurerepo)? If so, I suppose no harm in being a little more verbose.

Reply:
Hi @stmcginnis, are you asking why `contents: read` is repeated at workflow level and job level?

The `golangci-lint` job needs two permissions. If `contents: read` is not specified at the job level, it will not be granted to the job. GitHub permission model works in a way that if no permissions are specified, the top level ones will apply. But if they are specified at the job level, then the specific ones will apply. So if only `pull-requests: read` is specified at job level, only that will be granted. 

Since this is a public repo, that should also work. The tool that generates these fixes does not distinguish between a public or a private repo...so it ends up being a bit verbose for public repos...